### PR TITLE
sci-mathematics/octave: revision update and version bump to 7.1.0

### PR DESCRIPTION
**Pull request by ThamognyaKodi**
- uses bug: 839285 patch
- uses bug: 839285 revision update for 6.4.0-r2
- Bumps version 7.1.0

Signed-off-by: Thamognya Kodi <contact@thamognya.com>
Bug: https://bugs.gentoo.org/839285

### CODE REVIEW

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		sci-libs/umfpack:0=
+	)
+	ssl? (
+		dev-libs/openssl:0=
+	)
+	sundials? ( >=sci-libs/sundials-4:0= )
+	X? ( x11-libs/libX11:0= )"
+DEPEND="${RDEPEND}
+	dev-util/gperf
+	virtual/pkgconfig
```
Comment (thesamesam) [inline sci-mathematics/octave/octave-7.1.0.ebuild:79]: BDEPEND

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+		sci-libs/cxsparse:0=
+		sci-libs/umfpack:0=
+	)
+	ssl? (
+		dev-libs/openssl:0=
+	)
+	sundials? ( >=sci-libs/sundials-4:0= )
+	X? ( x11-libs/libX11:0= )"
+DEPEND="${RDEPEND}
+	dev-util/gperf
```
Comment (thesamesam) [inline sci-mathematics/octave/octave-7.1.0.ebuild:78]: I think BDEPEND

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+	virtual/pkgconfig
+	doc? (
```
Comment (thesamesam) [inline sci-mathematics/octave/octave-7.1.0.ebuild:80]: doc => BDEPEND

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+		dev-texlive/texlive-fontsrecommended
+		dev-texlive/texlive-plaingeneric
+		dev-texlive/texlive-metapost
+		virtual/latex-base
+	)
+	gui? ( dev-qt/linguist-tools:5 )
```
Comment (thesamesam) [inline sci-mathematics/octave/octave-7.1.0.ebuild:86]: BDEPEND

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+	java? ( >=virtual/jdk-1.6.0 )
+	qrupdate? ( app-misc/pax-utils )
+	sparse? ( app-misc/pax-utils )
+	|| ( media-gfx/imagemagick media-gfx/graphicsmagick[imagemagick] )"
```
Comment (thesamesam) [inline sci-mathematics/octave/octave-7.1.0.ebuild:90]: BDEPEND

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+		doins doc/interpreter/macros.texi
+	fi
+	[[ -e test/fntests.log ]] && dodoc test/fntests.log
+	use java && \
+		java-pkg_regjar "${ED}/usr/share/${PN}/${PV}/m/java/octave.jar"
+	echo "LDPATH=${EPREFIX}/usr/$(get_libdir)/${PN}/${PV}" > 99octave || die
+	doenvd 99octave
+}
+
+pkg_postinst() {
```
Comment (thesamesam) [inline sci-mathematics/octave/octave-7.1.0.ebuild:197]: Oh, xdg.eclass instead, maybe?

File: sci-mathematics/octave/files/octave-6.4.0-omit-qtchooser-qtver.patch [hunk 7]
```diff
@@ -0,0 +1,63 @@
+diff -urNp octave-6.4.0-orig/configure octave-6.4.0-dwok/configure
```
Comment (thesamesam) [inline sci-mathematics/octave/files/octave-6.4.0-omit-qtchooser-qtver.patch:1]: This is still only patching configure, not configure.ac, which is fragile -- and we're already calling eautoreconf anyway.

Also, given this patch is applied before eautoreconf (which is right), the changes are likely to get blown away, so this does not do anything right now, I think.

Just change it to only patch configure.ac then test it.

File: sci-mathematics/octave/octave-7.1.0.ebuild (hunk 6 above)
Comment (thesamesam) [reply sci-mathematics/octave/octave-7.1.0.ebuild:203]: ...?

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+		dev-qt/qtgui:5
+		dev-qt/qthelp:5
+		dev-qt/qtnetwork:5
+		dev-qt/qtopengl:5
+		dev-qt/qtprintsupport:5
+		dev-qt/qtwidgets:5
+		x11-libs/qscintilla:=
+	)
+	hdf5? ( sci-libs/hdf5:0= )
+	java? ( >=virtual/jre-1.6.0:* )
```
Comment (vaukai) [inline sci-mathematics/octave/octave-7.1.0.ebuild:51]: java? ( >=virtual/jre-1.8:* )

Same for 6.4.0-r2
See https://wiki.gentoo.org/wiki/Java_Developer_Guide#Dependencies

File: sci-mathematics/octave/octave-7.1.0.ebuild [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+	dev-util/gperf
+	virtual/pkgconfig
+	doc? (
+		dev-texlive/texlive-fontsrecommended
+		dev-texlive/texlive-plaingeneric
+		dev-texlive/texlive-metapost
+		virtual/latex-base
+	)
+	gui? ( dev-qt/linguist-tools:5 )
+	java? ( >=virtual/jdk-1.6.0 )
```
Comment (vaukai) [inline sci-mathematics/octave/octave-7.1.0.ebuild:93]: java? ( >=virtual/jdk-1.8:* )

Same for 6.4.0-r2
See https://wiki.gentoo.org/wiki/Java_Developer_Guide#Dependencies

**Comment by thesamesam** [comment]
ping

**Pull request closed without merging**